src/Nethermind/Nethermind.Core/Collections/DictionaryExtensions.cs

```diff
@@ -0,0 +1,14 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Collections.Generic;
+
+namespace Nethermind.Core.Collections;
+
+public static class DictionaryExtensions
+{
+    public static void Increment<TKey>(this IDictionary<TKey, int> dictionary, TKey key)
+    {
+        if (!dictionary.TryAdd(key, 1)) dictionary[key]++;
+    }
+}
```

Comment:
This doesn't guarantee thread-safety.

Maybe instead of using dictionaries in some places, we could use Arrays of ints, get ref to the array element and use Interlocked? If array indexes would be mapped to enum values, we could get labels with some attribute?

Reply:
This is not for the concurrent dictionary or the metric. This is for the counters in filtering candidate peers.

Reply:
If they are regular dictionaries you would get better results using [GetValueRefOrAddDefault](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal.getvaluereforadddefault?view=net-9.0#system-runtime-interopservices-collectionsmarshal-getvaluereforadddefault-3(system-collections-generic-dictionary((-0-1))-alternatelookup((-2))-2-system-boolean@)) and then incrementing the ref, we already use it in some places in our codebase